File: src/app/interfaces/job-application.ts
export interface JobApplication {
  id: number;
  fullName: string;
  email: string;
  phoneNumber: string;
  resume: string;
  country: string;
  city: string;
  jobTitle: string;
  jobDescription: string;
  jobApplicationStatus: string;
  dateCreated: string; 
  lettreMotivation: string;


  jobId: number;
  jobTitleFromJob: string;

  candidatId: number;
  candidatFullName: string;
}

File: src/main/java/sen/saloum/JobConnect/model/JobApplication.java
package sen.saloum.JobConnect.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.internal.build.AllowNonPortable;
import org.springframework.web.multipart.MultipartFile;

import java.time.OffsetDateTime;

@Entity
@Data
@NoArgsConstructor
public class JobApplication {
    @Id
    @GeneratedValue
    private Long id;
    private String fullName;
    private String email;
    private String phoneNumber;
    private String resume;
    private String country;
    private String city;
    private String jobTitle;
    private String jobDescription;
    private String jobApplicationStatus;
    private OffsetDateTime dateCreated;

    @ManyToOne
    @JoinColumn(name = "job_id")
    private Job job;
    @ManyToOne
    @JoinColumn(name = "candidat_id")
    private Candidat candidat;
}
